В проект добавлен Swagger

timurash opened the pull request:
Swagger находится по адресу: https://localhost:26038/swagger/index.html

https://github.com/timurash/timeTrackingService/issues/6

timurash committed:
В проект добавлен Swagger

https://github.com/timurash/timeTrackingService/issues/6

AlexandrYZ commented on PL/Controllers/ReportController.cs:
```diff
@@ -122,8 +129,9 @@ public IActionResult DeleteReport(int? id)
+        /// <response code="200">Если отчеты успешно получены.</response>
+        /// <response code="400">Если произошла неизвестная ошибка.</response>   
         [HttpGet("get")]
         public IActionResult Get([FromQuery] ReportFilterDTO reportFilterDTO)
```
Не указан тип возвращаемого ответа. Необходимо использовать ProducesResponseType.

AlexandrYZ replied:
Вижу что добавлены возможные http status коды, но не указан тип данных который возвращается при статусе 200.
Пример:
![image](https://user-images.githubusercontent.com/722891/90789736-af6f7f80-e320-11ea-81d4-36365d2d32ca.png)

AlexandrYZ commented:
Необходимо приложить скриншоты того как выглядит результат.

AlexandrYZ commented:
Необходимо обновить ветку.

timurash committed:
Добавлены атрибуты ProducesResponseType

timurash commented:
Скриншоты Swagger UI
<img width="1354" alt="Screen Shot 2020-08-16 at 22 19 20" src="https://user-images.githubusercontent.com/63206211/90340138-6e6d2780-e00f-11ea-9215-26f0dacd9ea7.png">
<img width="1371" alt="Screen Shot 2020-08-16 at 22 19 11" src="https://user-images.githubusercontent.com/63206211/90340145-7200ae80-e00f-11ea-80f5-a195f685b2f6.png">
<img width="1355" alt="Screen Shot 2020-08-16 at 22 19 03" src="https://user-images.githubusercontent.com/63206211/90340146-72994500-e00f-11ea-9ed2-5f8e365f0511.png">
<img width="1332" alt="Screen Shot 2020-08-16 at 22 18 52" src="https://user-images.githubusercontent.com/63206211/90340147-7331db80-e00f-11ea-8664-7c7870802022.png">
<img width="1462" alt="Screen Shot 2020-08-16 at 22 18 41" src="https://user-images.githubusercontent.com/63206211/90340148-73ca7200-e00f-11ea-9836-3ab38d2b7aa1.png">

AlexandrYZ commented:
Отсутствует описание полей на русском языке.
![image](https://user-images.githubusercontent.com/722891/90789000-ceb9dd00-e31f-11ea-8c32-1901cf914803.png)

AlexandrYZ commented:
В описании к задаче было бы полезно указать по какому адресу находится Swagger.

timurash committed:
В Swagger добавлены описания полей DTO классов

Проблема была в том, что в инициализации Swagger добавлялся только XML-файл уровня представления данных.
Теперь XML файлы уровня бизнес-логики и уровня доступа к данным так же добавлены в Swagger.

https://github.com/timurash/timeTrackingService/pull/11#issuecomment-677719155

timurash committed:
В ProducesResponseType добавлен тип данных, который возвращается в результате запроса

https://github.com/timurash/timeTrackingService/pull/11#discussion_r474055736

AlexandrYZ commented on PL/Controllers/UserController.cs:
```diff
@@ -141,8 +141,8 @@ public IActionResult DeleteUser(int? id)
         [HttpGet("get")]
-        [ProducesResponseType(StatusCodes.Status200OK)]
-        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(OkObjectResult), StatusCodes.Status200OK)]
```
В данном случае необходимо использовать `[ProducesResponseType(typeof(List<UserViewModel>), StatusCodes.Status200OK)]` потому что метод возвращает список пользователей:
```
var users = mapper.Map<IEnumerable<UserDTO>, List<UserViewModel>>(userDTOs);
return Ok(users);
```

После этого Swagger будет отображать правильную документацию для возвращаемого типа.

timurash committed:
ProducesResponeType теперь только на Get методах и возвращает правильный тип

https://github.com/timurash/timeTrackingService/pull/11#discussion_r474845616

timurash committed:
Обновление ветки до состояния master
